pkg/ingester/shipper.go

```diff
@@ -64,22 +70,30 @@ func newMetrics(reg prometheus.Registerer) *metrics {
 	return &m
 }
 
+type ConfigProvider interface {
+	OutOfOrderBlocksExternalLabelEnabled(userID string) bool
+}
+
 // Shipper watches a directory for matching files and directories and uploads
 // them to a remote data store.
 // Shipper implements BlocksUploader interface.
 type Shipper struct {
-	logger  log.Logger
-	dir     string
-	metrics *metrics
-	bucket  objstore.Bucket
-	source  metadata.SourceType
+	logger      log.Logger
+	cfgProvider ConfigProvider
+	userID      string
```

Comment:
[nit] To avoid the need for shipper to know the user, we could pass a function to shipper `func addOooLabel() bool` instead.

Reply:
We thought about this but then we saw how the compactor did it and the compactor passes the userid so I figured it would be best to have a similar approach across the different implementations.

Reply:
Sure, this works too. Let's keep it.